lua/apisix/plugins/response-rewrite.lua

```diff
@@ -0,0 +1,106 @@
+--
+-- Licensed to the Apache Software Foundation (ASF) under one or more
+-- contributor license agreements.  See the NOTICE file distributed with
+-- this work for additional information regarding copyright ownership.
+-- The ASF licenses this file to You under the Apache License, Version 2.0
+-- (the "License"); you may not use this file except in compliance with
+-- the License.  You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+--
+local core        = require("apisix.core")
+local plugin_name = "response-rewrite"
+local ngx         = ngx
+local pairs       = pairs
+
+
+local schema = {
+    type = "object",
+    properties = {
+        headers = {
+            description = "new headers for repsonse",
+            type = "object",
+            minProperties = 1,
+        },
+        body = {
+            description = "new body for repsonse",
+            type = "string",
+        },
+        status_code = {
+            description = "new status code for repsonse",
+            type = "number",
+            minimum = 200,
+            maximum = 598,
+        }
+    },
+    minProperties = 1,
+}
+
+
+local _M = {
+    version  = 0.1,
+    priority = 899,
+    name     = plugin_name,
+    schema   = schema,
+}
+
+
+function _M.check_schema(conf)
+    local ok, err = core.schema.check(schema, conf)
+    if not ok then
+        return false, err
+    end
+
+    --reform header from object into array, so can avoid use pairs, which is NYI
+    if conf.headers then
+        conf.headers_arr = {}
+        for field, value in pairs(conf.headers) do
+            core.table.insert(conf.headers_arr, field)
+            core.table.insert(conf.headers_arr, value)
+        end
+    end
+
+    return true
+end
+
+
+do
+
+function _M.body_filter(conf, ctx)
+    if conf.body then
+        if ngx.arg[2] then
```

Comment:
When the response body is multiple chunks, we actually only want the first one to be overwritten, and the second one to set the end symbol.

Reply:
I think is easier to check the last one（by check the `ngx.arg[2]`） and overwrite the fore chunks, , if we want the first chunk to be overwritten, then I should mark the non-first chunk. that make things complicated

Reply:
or I could use ngx.ctx mark it  to drop the non-first chunk

Reply:
as far as I know ,When setting nil or an empty Lua string value to ngx.arg[1], no data chunk will be passed to the downstream Nginx output filters at all.

Reply:
How about this style?

```lua
    if conf.body then
        ngx.arg[1] = conf.body
        ngx.arg[2] = true
    end
```

Reply:
greate